packages/vite/src/node/plugins/worker.ts

```diff
@@ -55,9 +55,11 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
               format: 'es',
               sourcemap: config.build.sourcemap
             })
-            url = `data:application/javascript;base64,${Buffer.from(
-              output[0].code
-            ).toString('base64')}`
+            
+            return `export default function WorkerWrapper() {
+              const blob = new Blob([atob(\"${Buffer.from(output[0].code).toString('base64')}\")], { type: 'text/javascript;charset=utf-8' });
+              return new Worker((window.URL || window.webkitURL).createObjectURL(blob));
```

Comment:
`createObjectURL` creates a pointer to the blob. I believe it can be freed after creating the Worker with `revokeObjectURL`.

Reply:
Oh I didn't realize it required some manual memory management. In my project this is completely unnecessary since I only have long-lived webworkers, thus irrelevant to my use case. I guess one change I can do is just move the objectURL outside of that function, so that it doesn't recreate it on multiple calls.

I'm really curious to know how one would solve this issue in a automated way.

Reply:
I believe the suggestion is to do (entirely untested)
```ts
const blob = new Blob([atob(\"${Buffer.from(output[0].code).toString('base64')}\")], { type: 'text/javascript;charset=utf-8' }); 
const objURL = (window.URL || window.webkitURL).createObjectURL(blob);
const worker = new Worker(obj);
(window.URL || window.webkitURL).revokeObjectURL(blob);
return worker;
```

Reply:
Yup, immediately revoking the URL as shown above will fix the issue. 

Slight change in the snippet: Pass the URL to revoke, not the blob
```diff
const objURL = (window.URL || window.webkitURL).createObjectURL(blob);
const worker = new Worker(obj);
-(window.URL || window.webkitURL).revokeObjectURL(blob);
+(window.URL || window.webkitURL).revokeObjectURL(objURL);
```